src/utils/profile.ts

```diff
@@ -0,0 +1,121 @@
+import { Profile, ProfileLink, ResponseData3Box } from '../models/Profile'
+import axios, { AxiosResponse, CancelToken } from 'axios'
+import jwtDecode from 'jwt-decode'
+import { Logger } from '@oceanprotocol/lib'
+
+// https://docs.3box.io/api/rest-api
+const apiUri = 'https://ipfs.3box.io'
+const ipfsUrl = 'https://ipfs.oceanprotocol.com'
+
+function decodeProof(proofJWT: string) {
+  if (!proofJWT) return
+  const proof = jwtDecode(proofJWT) as any
+  return proof
+}
+
+function getLinks(
+  website: string,
+  twitterProof: string,
+  githubProof: string
+): ProfileLink[] {
+  // Conditionally add links if they exist
+  const links = [
+    ...(website ? [{ name: 'Website', value: website }] : []),
+    ...(twitterProof
+      ? [
+          {
+            name: 'Twitter',
+            value: decodeProof(twitterProof).claim.twitter_handle
+          }
+        ]
+      : []),
+    ...(githubProof
+      ? [{ name: 'GitHub', value: githubProof.split('/')[3] }]
+      : [])
+  ]
+
+  return links
+}
+
+function transformResponse({
+  name,
+  description,
+  website,
+  emoji,
+  image,
+  /* eslint-disable camelcase */
+  proof_twitter,
+  proof_github,
+  proof_did
+}: ResponseData3Box) {
+  /* eslint-enable camelcase */
+  const links = getLinks(website, proof_twitter, proof_github)
+
+  const profile: Profile = {
+    did: decodeProof(proof_did).iss,
+    // Conditionally add profile items if they exist
+    ...(name && { name }),
+    ...(description && { description }),
+    ...(emoji && { emoji }),
+    ...(image && {
+      image: `${ipfsUrl}/ipfs/${
+        image.map(
+          (img: { contentUrl: { [key: string]: string } }) =>
+            img.contentUrl['/']
+        )[0]
+      }`
+    }),
+    ...(links.length && { links })
+  }
+
+  return profile
+}
+
+export default async function get3BoxProfile(
+  accountId: string,
+  cancelToken: CancelToken
+): Promise<Profile> {
+  try {
+    // const interceptor = axios.interceptors.response.use(
+    //   (response) => {
+    //     // Any status code that lie within the range of 2xx cause this function to trigger
+    //     // Do something with response data
+    //     return response
+    //   },
+    //   (error) => {
+    //     // Any status codes that falls outside the range of 2xx cause this function to trigger
+    //     // Do something with response error
+    //     console.log('Intercepted error', error.message)
+    //     return Promise.resolve({ status: 200 })
+    //   }
+    // )
+
+    const response: AxiosResponse<ResponseData3Box> = await axios(
+      `${apiUri}/profile?address=${accountId}`,
+      {
+        cancelToken
+        // validateStatus: () => true
+      }
+    )
+    // axios.interceptors.request.eject(interceptor)
+
+    // TODO: prevent 404 from showing up in console somehow.
```

Comment:
as expected, this is tricky. Cause the browser is doing that, not the library, and showing an error in console upon any `404` response is the default in most browsers. And we can't ship production code where by default `404` errors will be thrown.

I had 3 strategies so far:
1. silent error `catch`
2. make axios trick into resolving promise with any status code, and go silent from there
3. intercept the response with axios, and if `404`, go silent

But all of them can't get to the lower levels required to prevent a 404 being shown in browser console.

As far as I see, the only possibility would be to create our own quick API proxy which always returns a `200`. That's fine cause 3Box response sends `{ "status": "error" }` when no profile is found. So we just need another response status code which does not trigger browser errors. 

Any other ideas?

Reply:
The API proxy would be the best approach, 1 or 3 look interesting, maybe less effort

Reply:
What I meant was that I already tried option 1-3, and they are not working, and option 4 is too obtrusive. So we have this proxy now: https://github.com/oceanprotocol/market-stats/blob/main/api/profile.ts

Reply:
Oh, i misunderstood then, my bad.